set.go

```diff
@@ -438,3 +438,38 @@ func (s *Set) registerMetric(name string, m metric) {
 		panic(fmt.Errorf("BUG: metric %q is already registered", name))
 	}
 }
+
+// UnregisterMetric will remove a registered metric
+func (s *Set) UnregisterMetric(name string) {
+	if err := validateMetric(name); err != nil {
```

Comment:
I think there is no need in validating the metric when unregistering it. I'd change the `UnregisterMetric` API to return `true` if the metric has been successfully unregistered or `false` if there was no such metric in the `Set`.

Reply:
Agreed. I will change that.